src/Mod/PartDesign/SprocketFeature.py

```diff
@@ -0,0 +1,232 @@
+#*   as published by the Free Software Foundation; either version 2 of     *
+#*   the License, or (at your option) any later version.                   *
+#*   for detail see the LICENCE text file.                                 *
+#*                                                                         *
+#*   This program is distributed in the hope that it will be useful,       *
+#*   but WITHOUT ANY WARRANTY; without even the implied warranty of        *
+#*   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the         *
+#*   GNU Library General Public License for more details.                  *
+#*                                                                         *
+#*   You should have received a copy of the GNU Library General Public     *
+#*   License along with this program; if not, write to the Free Software   *
+#*   Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  *
+#*   USA                                                                   *
+#*                                                                         *
+#***************************************************************************
+
+import FreeCAD, Part
+from fcsprocket import fcsprocket
+from fcsprocket import sprocket
+
+if FreeCAD.GuiUp:
+    import FreeCADGui
+    from PySide import QtCore, QtGui
+    from FreeCADGui import PySideUic as uic
+
+__title__="PartDesign SprocketObject management"
+__author__ = "Adam Spontarelli"
+__url__ = "http://www.freecadweb.org"
+
+
+def makeSprocket(name):
+    '''makeSprocket(name): makes a Sprocket'''
+    obj = FreeCAD.ActiveDocument.addObject("Part::Part2DObjectPython",name)
+    _Sprocket(obj)
+    if FreeCAD.GuiUp:
+        _ViewProviderSprocket(obj.ViewObject)
+    #FreeCAD.ActiveDocument.recompute()
+    if FreeCAD.GuiUp:
+        body=FreeCADGui.ActiveDocument.ActiveView.getActiveObject("pdbody")
+        part=FreeCADGui.ActiveDocument.ActiveView.getActiveObject("part")
+        if body:
+            body.Group=body.Group+[obj]
+        elif part:
+            part.Group=part.Group+[obj]
+    return obj
+
+class _CommandSprocket:
+    "the Fem Sprocket command definition"
+    def GetResources(self):
+        return {'Pixmap'  : 'PartDesign_Sprocket',
+                'MenuText': QtCore.QT_TRANSLATE_NOOP("PartDesign_Sprocket","Sprocket..."),
+                'Accel': "",
+                'ToolTip': QtCore.QT_TRANSLATE_NOOP("PartDesign_Sprocket","Creates or edit the sprocket definition.")}
+        
+    def Activated(self):
+
+        FreeCAD.ActiveDocument.openTransaction("Create Sprocket")
+        FreeCADGui.addModule("SprocketFeature")
+        FreeCADGui.doCommand("SprocketFeature.makeSprocket('Sprocket')")
+        FreeCADGui.doCommand("Gui.activeDocument().setEdit(App.ActiveDocument.ActiveObject.Name,0)")
+        
+    def IsActive(self):
+        if FreeCAD.ActiveDocument:
+            return True
+        else:
+            return False
+
+       
+class _Sprocket:
```

Comment:
I suggest not using underscores for the main classes of the scripted objects. So, it should be `Sprocket` not `_Sprocket`. The fact that many classes have an underscore is actually a mistake, because then many automated systems of documentation like Doxygen will omit the class and the object will not documented.

I know our documentation doesn't exactly tell you this, but being honest, our documentation and examples aren't the best out there, and they are in sever need of an update. See [Why are class names prefixed with an underscore?](https://forum.freecadweb.org/viewtopic.php?f=23&t=44612)

Same with the view provider. It should be `ViewProviderSprocket`.

Reply:
What about SprocketTaskPanel?

Reply:
I would not use underscores for any of them.